src/test/java/seedu/address/logic/commands/UndoCommandTest.java

```diff
@@ -0,0 +1,178 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import java.util.Comparator;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.commands.results.CommandResult;
+import seedu.address.logic.commands.results.UndoCommandResult;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.item.Internship;
+import seedu.address.model.item.Person;
+import seedu.address.model.item.Project;
+import seedu.address.model.item.Resume;
+import seedu.address.testutil.InternshipBuilder;
+import seedu.address.testutil.PersonBuilder;
+import seedu.address.testutil.ResumeBuilder;
+import seedu.address.testutil.TypicalInternship;
+import seedu.address.testutil.TypicalResume;
+import seedu.address.testutil.TypicalResumeBook;
+
+
+public class UndoCommandTest {
+    private Model expectedModel = new ModelManager(TypicalResumeBook.TYPICAL_WITH_FILLED_RESUME, new UserPrefs());
+    private Model model = new ModelManager(TypicalResumeBook.TYPICAL_WITH_FILLED_RESUME, new UserPrefs());
+
+    @BeforeEach
+    public void setUp() {
+        expectedModel = new ModelManager(TypicalResumeBook.TYPICAL_WITH_FILLED_RESUME, new UserPrefs());
```

Comment:
I think it's safer to get a new copy every time, new ResumeBook(TypicalResumeBook.TYPICAL_WITH_FILLED_RESUME).build()

Reply:
@helloImHai did you mean new ResumeBook**Builder**(TypicalResumeBook.TYPICAL_WITH_FILLED_RESUME).build() ?

Reply:
YES hahahaha

Reply:
We do not need to, as the argument appears as `ReadOnlyResumeBook` interface, which only permits read and no modification.

Also the internal resume book is made fresh new within the constructor.